components/domains/externalDomainTransferModal.tsx

```diff
@@ -0,0 +1,168 @@
+import { InputAdornment, Modal, TextField } from "@mui/material";
+import { useContractWrite } from "@starknet-react/core";
+import { useRouter } from "next/router";
+import React, {
+  FunctionComponent,
+  useState,
+  useEffect,
+  useContext,
+} from "react";
+import styles from "../../styles/components/modalMessage.module.css";
+import { isHexString, minifyAddress } from "../../utils/stringService";
+import Button from "../UI/button";
+import { utils } from "starknetid.js";
+import { StarknetIdJsContext } from "../../context/StarknetIdJsProvider";
+import ConfirmationTx from "../UI/confirmationTx";
+import { useNotificationManager } from "../../hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "../../utils/constants";
+import { Call } from "starknet";
+
+type ExternalDomainsTransferModalProps = {
+  domain: string;
+  domainEncoded: string;
+  resolverContract: string;
+  handleClose: () => void;
+  isModalOpen: boolean;
+  domainKind?: string;
+};
+
+const ExternalDomainsTransferModal: FunctionComponent<
+  ExternalDomainsTransferModalProps
+> = ({
+  domainEncoded,
+  resolverContract,
+  handleClose,
+  isModalOpen,
+  domainKind,
+}) => {
+  const router = useRouter();
+  const { externalDomain: domain } = router.query;
+  const [targetAddress, setTargetAddress] = useState<string>("");
+  const { addTransaction } = useNotificationManager();
+  const [addressInput, setAddressInput] = useState<string>("");
+  const { starknetIdNavigator } = useContext(StarknetIdJsContext);
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [callData, setCallData] = useState<Call[]>([]);
+
+  const { writeAsync: transfer_name, data: transferData } = useContractWrite({
+    calls: callData,
+  });
+
+  useEffect(() => {
+    setCallData([
+      {
+        contractAddress: resolverContract,
+        entrypoint: "transfer_name",
+        calldata: [domainEncoded, targetAddress],
+      },
+    ]);
+  }, [domainEncoded, targetAddress, resolverContract]);
+
+  useEffect(() => {
+    if (isHexString(addressInput)) {
+      setTargetAddress(addressInput);
+    } else if (utils.isStarkDomain(addressInput)) {
+      starknetIdNavigator?.getAddressFromStarkName(addressInput).then((res) => {
+        if (!res || res === "0x0") setTargetAddress("");
+        else setTargetAddress(res);
+      });
+    } else {
+      setTargetAddress("");
+    }
+  }, [addressInput, starknetIdNavigator]);
+
+  useEffect(() => {
+    if (!transferData?.transaction_hash) return;
+    addTransaction({
+      timestamp: Date.now(),
+      subtext: `${domain} transfered`,
+      type: NotificationType.TRANSACTION,
+      data: {
+        type: TransactionType.TRANSFER_IDENTITY,
```

Comment:
it's actually not transfer of an identity here don't you think ?

Reply:
Oh and I see you created the good type `TRANSFER_EXTERNAL_DOMAIN` so yeah it's a forget

Reply:
Yes, I forgot to push that change